back-end/lms/views/absence.py

```diff
@@ -116,44 +162,122 @@ def put(self, request: Request) -> Response:
         :param request:
         :return:
         """
-        absence = AbsenceSerializer(data=request.data)
-        try:
-            if absence.is_valid():
-                absence = absence.save()
+        existing_absence = Absence.objects.filter(id=request.data['id'])
+        if existing_absence.exists():
+            absence_ser = AbsenceSerializer(data=request.data)
+            if absence_ser.is_valid():
+                absence_ser.update(instance=existing_absence,
+                                   validated_data=request.data)
                 return Response(
                     {
-                        'message': f'Absence record with id {absence.id} '
-                                   f'successfuly created'
+                        'message': f'Absence with id {request.data["id"]} '
+                                   f'successfully modified'
                     },
                     status=HTTP_200_OK)
-            return Response({'message': absence.errors},
+            return Response({'message': absence_ser.errors},
                             status=HTTP_400_BAD_REQUEST)
-        except IntegrityError:
+        return Response(
+            {
+                'message': f'Absence with id {request.data["id"]} '
+                           f'does not exist in this database'
+            },
+            status=HTTP_400_BAD_REQUEST)
+    
+    # pylint: disable=no-self-use
+    @csrf_exempt
+    def delete(self, request: Request) -> Response:
+        """
+        DELETE - function uses id from request 'query'
+        :param request:
+        :return:
+        """
+        absence_to_delete = Absence.objects.filter(
+            id=request.query_params['id'])
+        if absence_to_delete.exists():
+            absence_to_delete.delete()
             return Response(
                 {
-                    'message': 'A record with this student and this date '
-                               'already exists. Please, modify existing record.'
+                    'message': f'Absence with id {request.query_params["id"]} '
+                               f'successfully deleted'
                 },
-                status=HTTP_400_BAD_REQUEST)
+                status=HTTP_200_OK)
+        return Response(
+            {
+                'message': f'Absence with id {request.query_params["id"]} '
+                           f'does not exist in this database'
+            },
+            status=HTTP_400_BAD_REQUEST)
 
 
 @permission_classes((AllowAny,))
```

Comment:
This decorator works for function-based views only. For classes, you have to specify `permission_classes`:
```python
class AbsenceJournalView(APIView):
        permission_classes = [AllowAny]
```